src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -20,18 +21,24 @@ public abstract class Renderer : IDisposable
         private readonly Dictionary<int, ComponentState> _componentStateById = new Dictionary<int, ComponentState>();
         private readonly RenderBatchBuilder _batchBuilder = new RenderBatchBuilder();
         private readonly Dictionary<int, EventHandlerInvoker> _eventBindings = new Dictionary<int, EventHandlerInvoker>();
+        private RendererSynchronizationContext _synchronizationContext = new RendererSynchronizationContext();
```

Comment:
We probably want to receive this on the constructor and RendererSynchronizationContext might need to be public

Reply:
I agree - it would be great for this to be a constructor param, as per other comments. That would increase the flexibility a lot, e.g., being able to pass in a sync context owned by the CircuitHost.

Not sure why it would need to be public but you've probably thought this through more than me!

Reply:
Also then for `WebAssemblyRenderer`, assuming it's not passing in a `RendererSynchronizationContext`, that would allow the linker to strip out `RendererSynchronizationContext` and any dependency.

For `WebAssemblyRenderer` in particular, it would be good if it could have a `null` sync context so we don't accidentally try to `Invoke` on it in an asynchronous way (rather than using `SynchronizationContext.Invoke`, which calls into `ThreadPool.QueueUserWorkItem`), or if that makes the code messy with special cases, consider defining some interface like `ISynchronizationContextDispatcher` that can have a no-op passthrough implementation for the `WebAssemblyRenderer` case.

Reply:
I'll figure something out along those lines

Reply:
I created an IDispatcher interface, we don't set it up on the web assembly case (hence I removed the closure for dispatch) we simply check if the IDispatcher is a SynchronizationContext and throw if that's the case.

I also provided a CreateDefaultDispatcher static method so that it can be created externally (and the RendererSynchronizationContext becomes an implementation detail).

We need to do work for the linker to strip things in the WASM case, but it's totally doable.